plugin/trino-mysql/src/main/java/io/trino/plugin/mysql/MySqlClient.java

```diff
@@ -457,6 +465,13 @@ public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connect
                         timeWriteFunction(timeType.getPrecision())));
 
             case Types.TIMESTAMP:
+                if (enforceTimestampMillisPrecision) {
+                    return Optional.of(ColumnMapping.longMapping(
+                            TIMESTAMP_MILLIS,
+                            timestampMillisReadFunction(),
+                            timestampWriteFunction(TIMESTAMP_MILLIS),
+                            DISABLE_PUSHDOWN));
+                }
                 TimestampType timestampType = createTimestampType(getTimestampPrecision(typeHandle.getRequiredColumnSize()));
                 checkArgument(timestampType.getPrecision() <= TimestampType.MAX_SHORT_PRECISION, "Precision is out of range: %s", timestampType.getPrecision());
```

Comment:
Can we simply override what we are getting from `timestampType.getPrecision()`?

Reply:
I wasnt sure on the FULL_PUSHDOWN PredicatePushdownController. for Millis, the pushdown was disabled.

Reply:
Pushdown can be tricky here. Make sure you test it. Should we disable it if precision is overriden?

Reply:
Pushdown can be tricky here. Make sure you test it. Should we disable it if precision is overriden?

Reply:
Pushdown can be tricky here. Make sure you test it. Should we disable it if precision is overriden?